Modules/Sources/DesignSystem/Components/DSEditableListDetailItem.swift

```diff
@@ -0,0 +1,26 @@
+import SwiftUI
+
+public struct DSEditableListDetailItemView: View {
```

Comment:
I'm skipping reviewing the `ViewLayer` part in this PR. 

Regarding the `DS` part, let's not extend the `DesignSystem` framework and let's not use the `DS` prefix moving forward. The plan is to remove the `DesignSystem` framework along with all of its APIs. The `DSEditableListDetailItemView` and  `DSListDetailItem` components are also arguably too simply to be defined as shared components. They look like specific implementations needed for the `ApplicationTokenListView.

Reply:
I'm not aware of our plan for `DesignSystem`, and what should/shouldn't go into it. @jkmassel Do you mind having a look? Should the new SwiftUI views be kept in the `DesignSystem` module, or somewhere else?